### PR TITLE
xtensa: Add 8ulp_adsp toolchain

### PR DESCRIPTION
**Pull request by iuliana-prodan**
Add Xtensa toolchain for the HiFi4 DSP from i.MX8ULP.

### CODE REVIEW

**Comment by iuliana-prodan** [comment]
@nashif @dcpleung can you, please, review.
Thanks!

File: overlays/xtensa_nxp_imx8ulp_adsp/gdb/gdb/xtensa-config.c
Comment (dcpleung) [inline overlays/xtensa_nxp_imx8ulp_adsp/gdb/gdb/xtensa-config.c:421]: You need to add this line:

```
xtensa_gdbarch_tdep xtensa_tdep (rmap);
```

in this file. Or else GDB would not build.

**Review by dcpleung — CHANGES_REQUESTED** [review]

File: overlays/xtensa_nxp_imx8ulp_adsp/gdb/gdb/xtensa-config.c
Comment (iuliana-prodan) [reply overlays/xtensa_nxp_imx8ulp_adsp/gdb/gdb/xtensa-config.c:421]: Done.

**Review by dcpleung — APPROVED** [review]

**Review by dbaluta — APPROVED** [review]

**Comment by dbaluta** [comment]
Looks good to me.

Cc: @mmahadevan108

**Comment by dbaluta** [comment]
@dcpleung @nashif I might be missing something but shouldn't the new toolchain be generated by CI? So that we can have a look and verify that all is OK?

**Comment by dcpleung** [comment]
IIRC, due to security issue, the CI does not generate a new toolchains from PRs, but only on main branch once PRs are merged.

**Comment by iuliana-prodan** [comment]
@dcpleung  @nashif @dleach02 can you please help us with this - we need the toolchain since we have some PRs dependent on this:
https://github.com/zephyrproject-rtos/zephyr/pull/63751
https://github.com/zephyrproject-rtos/hal_xtensa/pull/25
What can we do for the toolchain to be generated and test our PRs with it?
Thank you!

**Review by dbaluta — APPROVED** [review]

**Comment by dcpleung** [comment]
This will need to be merged first for CI to build it. I don't have merge right. @nashif or @stephanosio should be able to merge.

**Review by nashif — APPROVED** [review]

**Review by stephanosio — COMMENTED** [review]
@iuliana-prodan ~~It looks like there is a problem with GDB in this toolchain:~~ Nevermind. I just found out https://github.com/zephyrproject-rtos/sdk-ng/pull/712 is fixing the issue.

```
[INFO ]  =================================================================
[INFO ]  Installing cross-gdb
[EXTRA]    Configuring cross gdb
[EXTRA]    Building cross gdb
Error: [ERROR]           _bfd_error_handler (_("%pB: error: attribute section '%pA' too big: %#llx"),
Error: [ERROR]           _bfd_error_handler (_("%pB: error: attribute section '%pA' too big: %#llx"),
Error:     ../gnulib/import/wchar.h:747:1: error: expected ',' or '...' before 'pwc'
Error:     ../gnulib/import/wchar.h:813:1: error: expected ',' or '...' before 'dest'
Error:     ../gnulib/import/wchar.h:817:19: error: nonnull argument with out-of-range operand number (argument 1, operand 2)
Error:     ../gnulib/import/stdio.h:1340:1: error: expected ',' or '...' before 'lineptr'
Error:     ../gnulib/import/stdio.h:1344:19: error: nonnull argument with out-of-range operand number (argument 2, operand 2)
Error:     ../gnulib/import/stdio.h:1372:1: error: expected ',' or '...' before 'lineptr'
Error:     ../gnulib/import/stdio.h:1375:19: error: nonnull argument with out-of-range operand number (argument 2, operand 2)
Error:     ../gnulib/import/stdlib.h:1612:1: error: expected ',' or '...' before 'name'
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_vecs.h:27:20: error: 'gdb' was not declared in this scope
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_vecs.h:27:48: error: template argument 1 is invalid
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_vecs.h:27:48: error: template argument 2 is invalid
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_vecs.h:33:16: error: 'gdb' was not declared in this scope
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_vecs.h:33:44: error: template argument 1 is invalid
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_vecs.h:33:44: error: template argument 2 is invalid
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_vecs.h:33:16: error: 'gdb' was not declared in this scope
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_vecs.h:33:44: error: template argument 1 is invalid
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_vecs.h:33:44: error: template argument 2 is invalid
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_vecs.h:33:16: error: 'gdb' was not declared in this scope
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_vecs.h:33:44: error: template argument 1 is invalid
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_vecs.h:33:44: error: template argument 2 is invalid
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_vecs.h:33:9: error: variable or field 'dirnames_to_char_ptr_vec_append' declared void
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_vecs.h:33:16: error: 'gdb' was not declared in this scope
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_vecs.h:33:44: error: template argument 1 is invalid
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_vecs.h:33:44: error: template argument 2 is invalid
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_vecs.h:33:48: error: 'vecp' was not declared in this scope
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_vecs.h:33:54: error: expected primary-expression before 'const'
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_vecs.h:41:20: error: 'gdb' was not declared in this scope
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_vecs.h:41:48: error: template argument 1 is invalid
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_vecs.h:41:48: error: template argument 2 is invalid
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb-hashtab.h:34:14: error: 'unique_ptr' in namespace 'std' does not name a template type
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_optional.h:197:5: error: there are no arguments to 'gdb_assert' that depend on a template parameter, so a declaration of 'gdb_assert' must be available [-fpermissive]
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/print-utils.h:30:25: error: 'ULONGEST' was not declared in this scope
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/print-utils.h:35:24: error: 'LONGEST' was not declared in this scope
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/print-utils.h:40:20: error: 'ULONGEST' was not declared in this scope
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/print-utils.h:40:32: error: expected primary-expression before 'int'
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/print-utils.h:40:44: error: expression list treated as compound expression in initializer [-fpermissive]
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/print-utils.h:45:23: error: 'ULONGEST' was not declared in this scope
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/print-utils.h:45:35: error: expected primary-expression before 'int'
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/print-utils.h:45:47: error: expression list treated as compound expression in initializer [-fpermissive]
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/print-utils.h:50:26: error: 'LONGEST' was not declared in this scope
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/print-utils.h:57:33: error: 'LONGEST' was not declared in this scope
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/print-utils.h:57:46: error: expected primary-expression before 'int'
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/print-utils.h:57:55: error: expression list treated as compound expression in initializer [-fpermissive]
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/print-utils.h:66:26: error: 'LONGEST' was not declared in this scope
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/print-utils.h:66:39: error: expected primary-expression before 'int'
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/print-utils.h:66:50: error: expected primary-expression before 'int'
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/print-utils.h:66:65: error: expected primary-expression before 'int'
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/print-utils.h:67:5: error: expected primary-expression before 'int'
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/print-utils.h:67:21: error: expression list treated as compound expression in initializer [-fpermissive]
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/print-utils.h:71:47: error: 'CORE_ADDR' does not name a type; did you mean 'CORE_ADDR_MAX'?
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/print-utils.h:73:50: error: 'CORE_ADDR' does not name a type; did you mean 'CORE_ADDR_MAX'?
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_obstack.h:31:18: error: 'IsMallocable' was not declared in this scope
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_obstack.h:31:32: error: expected primary-expression before '>' token
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_obstack.h:31:35: error: '::value' has not been declared
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_obstack.h:42:18: error: 'IsMallocable' was not declared in this scope
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_obstack.h:42:32: error: expected primary-expression before '>' token
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_obstack.h:42:35: error: '::value' has not been declared
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_obstack.h:71:29: error: 'xmalloc' was not declared in this scope
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_obstack.h:72:28: error: 'xfree' was not declared in this scope
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/gdbtypes.h:81:44: error: 'hex_string' cannot be used as a function
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/gdbtypes.h:2833:8: error: 'htab_up' does not name a type; did you mean 'htab_t'?
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/completer.h:[444](https://github.com/zephyrproject-rtos/sdk-ng/actions/runs/6845783702/job/18611511495#step:14:445):3: error: 'htab_up' does not name a type; did you mean 'htab_t'?
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/completer.h:397:27: error: 'm_entries_hash' was not declared in this scope
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_optional.h:197:16: error: 'gdb_assert' was not declared in this scope
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_optional.h:197:16: error: 'gdb_assert' was not declared in this scope
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_optional.h:197:16: error: 'gdb_assert' was not declared in this scope
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_optional.h:197:16: error: 'gdb_assert' was not declared in this scope
Error:     /__w/_temp/workspace/build/.build/HOST-x86_64-w64-mingw32/xtensa-nxp_imx8ulp_adsp_zephyr-elf/src/gdb/gdb/../gdbsupport/gdb_optional.h:197:16: error: 'gdb_assert' was not declared in this scope
Error:     make[3]: *** [xtensa-config.o] Error 1
Error:     make[3]: *** Waiting for unfinished jobs....
Error:     make[2]: *** [all-gdb] Error 2
Error:     make[1]: *** [all] Error 2
```

https://github.com/zephyrproject-rtos/sdk-ng/actions/runs/6845783702/job/18611511495#step:14:438

**Pull request merged**
